tests/unit/test_repository.py

```diff
@@ -245,107 +250,245 @@ def test_create_pull_request(repository_client: Client, mock_pull_request: PullR
+
+    with pytest.raises(RepositoryClientError) as exc:
+        repository_client.get_tag_file_content(path="path 1", tag_name="tag 1")
+
+    assert_substrings_in_string(("unauthorized", "401"), str(exc.value).lower())
+
+
+def test_get_tag_file_content_tag_unknown_object_error(
+    monkeypatch: pytest.MonkeyPatch, repository_client: Client
+):
+    """
+    arrange: given Client with a mocked github repository client that raises an
+        UnknownObjectException exception during tag operations
+    act: when get_tag_file_content is called
     assert: RepositoryFileNotFoundError is raised.
     """
     mock_github_repository = mock.MagicMock(spec=Repository)
-    mock_github_repository.get_contents.side_effect = [
-        UnknownObjectException(status=404, data="File not found error", headers=None)
-    ]
+    mock_github_repository.get_git_ref.side_effect = UnknownObjectException(
+        status=404, data="File not found error", headers=None
+    )
     monkeypatch.setattr(repository_client, "_github_repo", mock_github_repository)
-    path = "path 1"
+    tag_name = "tag 1"
 
-    with pytest.raises(RepositoryFileNotFoundError) as exc:
-        repository_client.get_file_content(path=path)
+    with pytest.raises(RepositoryTagNotFoundError) as exc:
+        repository_client.get_tag_file_content(path="path 1", tag_name=tag_name)
 
-    assert_substrings_in_string(("not", "retrieve", "file", path), str(exc.value).lower())
+    assert_substrings_in_string(("not", "retrieve", "tag", tag_name), str(exc.value).lower())
 
 
-def test_get_file_content_list(monkeypatch: pytest.MonkeyPatch, repository_client: Client):
+def test_get_tag_file_content_content_github_error(
+    monkeypatch: pytest.MonkeyPatch, repository_client: Client
+):
     """
-    arrange: given Client with a mocked github repository client that returns a list
-    act: when get_file_content is called
+    arrange: given Client with a mocked github repository client that raises an exception during
+        content operations
+    act: when get_tag_file_content is called
+    assert: RepositoryClientError is raised.
+    """
+    mock_github_repository = mock.MagicMock(spec=Repository)
+    mock_github_repository.get_contents.side_effect = GithubException(
+        status=401, data="unauthorized", headers=None
+    )
+    monkeypatch.setattr(repository_client, "_github_repo", mock_github_repository)
+
+    with pytest.raises(RepositoryClientError) as exc:
+        repository_client.get_tag_file_content(path="path 1", tag_name="tag 1")
+
+    assert_substrings_in_string(("unauthorized", "401"), str(exc.value).lower())
+
+
+def test_get_tag_file_content_unknown_object_error(
+    monkeypatch: pytest.MonkeyPatch, repository_client: Client
+):
+    """
+    arrange: given Client with a mocked github repository client that raises an
+        UnknownObjectException exception during content operations
```

Comment:
This test and `test_get_tag_file_content_tag_unknown_object_error` are very close each other, can we be more explicit in the "arrange" and specify which method raises the UnknownObjectException for both method ?

Reply:
the `arrange` already states `during tag operations` for one and `during content operations` for the other, the reason the method name isn't in the description is that it is possible to see it from the test code and the test description is a bit higher level without including the specific operation name

Reply:
Yes I saw that there was this very small difference, but it’s very difficult to catch the difference at a first sight.
Any idea on how to make the difference more explicit to avoid confusion ?

Reply:
the name of the tests have a difference too, I can't think of anything else

Reply:
Ok let’s keep it like this, that’s a small detail